mockkube/src/test/java/io/strimzi/test/mockkube3/MockKube3ControllersMockTest.java

```diff
@@ -133,8 +133,11 @@ public void testDeploymentController() {
                 .build();
 
         client.apps().deployments().inNamespace(namespace).resource(dep).create();
-
-        TestUtils.waitFor("Wait for deployment to have status", 100L, 10_000L, () -> client.apps().deployments().inNamespace(namespace).withName(deploymentName).get() != null && client.apps().deployments().inNamespace(namespace).withName(deploymentName).get().getStatus() != null);
+        
+        TestUtils.waitFor("Wait for deployment to have status", 100L, 10_000L, 
+                () -> client.apps().deployments().inNamespace(namespace).withName(deploymentName).get() != null
+                        && client.apps().deployments().inNamespace(namespace).withName(deploymentName).get().getStatus() != null
+                        && client.apps().deployments().inNamespace(namespace).withName(deploymentName).get().getStatus().getObservedGeneration() != null);
```

Comment:
I'm not sure what exactly was failing on this. But you are not anymore waiting for it to have non-empty status. So you should update the message.